### PR TITLE
Added upgrade-insecure-requests CSP directive

### PR DESCRIPTION
**Pull request by HarveyPeachey**
Resolves https://github.com/bbc/simorgh-infrastructure/issues/1025

**Overall change:**
Added [`upgrade-insecure-requests`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/upgrade-insecure-requests) CSP directive which instructs user agents (browsers), to treat HTTP URLs as HTTPS.

**Code changes:**

- Added upgrade-insecure-requests directive via helmet-csp.
- Updated unit test

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**

As this change applies the directive to our CSP headers, this might need heavy regression testing unless it's expected that all our URL's should be served via HTTPS, because if we have existing HTTP URL's that get forced into HTTPS then it might break if HTTPS is not supported by that URL.

_Cross-browser support_ - we know that not all browsers support this directive (see comments below) and some may have different implementations. When the directive was applied by Sport a couple of years ago we started seeing redirect loops in Firefox. We therefore need to be extra careful it behaves as expected across the range of browsers and devices we primarily support

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [x] This PR requires manual testing

### CODE REVIEW

File: src/server/utilities/cspHeader/index.js [hunk 1]
```diff
@@ -363,6 +363,7 @@ const helmetCsp = ({ isAmp, isLive }) => ({
     'style-src': generateStyleSrc({ isAmp, isLive }),
     'worker-src': generateWorkerSrc({ isAmp }),
     'report-to': 'default',
+    'upgrade-insecure-requests': true,
```
Comment (HarveyPeachey) [inline src/server/utilities/cspHeader/index.js:364]: Unfortunately, this directive is not supported by IE11 or Opera Mini https://caniuse.com/#feat=upgradeinsecurerequests. I have tested this with some includes on IE11 via BrowserStack and for some reason, they were behaving correctly as images were loading in with no console errors.

Comment (HarveyPeachey) [reply src/server/utilities/cspHeader/index.js:364]: The only other way I can think of supporting these browsers is to manipulate them during the render on the server by replacing `http` with `https` before it's served to the user.

Comment (andrewscfc) [reply src/server/utilities/cspHeader/index.js:364]: >  I have tested this with some includes on IE11 via BrowserStack and for some reason, they were behaving correctly as images were loading in with no console errors.

Did you test `latest` on browserstack to confirm the header you added actually made the images load in IE? I'm not entirely sure the `content-security-policy` header in general is supported in IE.

**Review by andrewscfc — APPROVED** [review]
LGTM, but just added a question clarify how this works on IE; in any case working on IE is probably not essential, that can be confirmed by @jimjohnsonrollings 

re:
>  if we have existing HTTP URL's that get forced into HTTPS then it might break if HTTPS is not supported by that URL

I don't believe this will be the case as our live csp has no insecure `http` urls in it, I believe this change would only be an improvement to broken requests that will now work correctly over `https`.

File: src/server/utilities/cspHeader/index.js (hunk 1 above)
Comment (HarveyPeachey) [reply src/server/utilities/cspHeader/index.js:364]: Ahh forgot about that will double-check with latest. I know that when I was looking around helmet-csp's source code it was doing browser sniffing to apply the different variations of `content-security-policy` to browsers, where IE11 uses `x-content-security-policy`. @andrewscfc

**Comment by jimjohnsonrollings** [comment]
> working on IE is probably not essential, that can be confirmed by @jimjohnsonrollings

Confirmed

File: src/server/utilities/cspHeader/index.js (hunk 1 above)
Comment (andrewscfc) [reply src/server/utilities/cspHeader/index.js:364]: You are correct @HarveyPeachey, but we don't whitelist the IE headers in mozart so they are never sent to the user on test and live. I imagine though for your browserstack test from local they _were_ present. No big deal, it's nice it works!

**Review by ghost — APPROVED** [review]
LGTM. 👍

**Comment by paruchurisilpa** [comment]
I have tested this and I have checked all the includes mentioned on the spreadsheet but the include did not load at all for any of the includes and throwing the following console errors:

Testing NEWS-6625 : (Include does not load)

- error - TypeError: NetworkError when attempting to fetch resource. 
- error - { "event": "toggle_fetch_error", "message": {} }
- Content Security Policy: The page's settings blocked the loading of a resource at https://logws1363.ati-host.net/?s=598342&idclient=86d1b7ac-b…5.page&r=1440x900x24x24&re=1432x393&hl=8x34x13&lng=en-GB&x1=[urn:bbc:cps:d1665420-b348-d54b-9e55-3766667eae59]
- Content Security Policy: The page's settings blocked the loading of a resource at https://toggles.test.api.bbci.co.uk/toggles?application=simo…_amp_source_origin=https://www.test.bbc.com&geoiplookup=true ("connect-src").

**Comment by HarveyPeachey** [comment]
> I have tested this and I have checked all the includes mentioned on the spreadsheet but the include did not load at all for any of the includes and throwing the following console errors:
> 
> Testing NEWS-6625 : (Include does not load)
> 
> * error - TypeError: NetworkError when attempting to fetch resource.
> * error - { "event": "toggle_fetch_error", "message": {} }
> * Content Security Policy: The page's settings blocked the loading of a resource at [https://logws1363.ati-host.net/?s=598342&idclient=86d1b7ac-b…5.page&r=1440x900x24x24&re=1432x393&hl=8x34x13&lng=en-GB&x1=[urn:bbc:cps:d1665420-b348-d54b-9e55-3766667eae59]](https://logws1363.ati-host.net/?s=598342&idclient=86d1b7ac-b%E2%80%A65.page&r=1440x900x24x24&re=1432x393&hl=8x34x13&lng=en-GB&x1=%5Burn:bbc:cps:d1665420-b348-d54b-9e55-3766667eae59%5D)
> * Content Security Policy: The page's settings blocked the loading of a resource at [https://toggles.test.api.bbci.co.uk/toggles?application=simo…_amp_source_origin=https://www.test.bbc.com&geoiplookup=true](https://toggles.test.api.bbci.co.uk/toggles?application=simo%E2%80%A6_amp_source_origin=https://www.test.bbc.com&geoiplookup=true) ("connect-src").
> 


Are those errors present on branch `latest` or only this branch?

**Comment by paruchurisilpa** [comment]
> > I have tested this and I have checked all the includes mentioned on the spreadsheet but the include did not load at all for any of the includes and throwing the following console errors:
> > Testing NEWS-6625 : (Include does not load)
> > 
> > * error - TypeError: NetworkError when attempting to fetch resource.
> > * error - { "event": "toggle_fetch_error", "message": {} }
> > * Content Security Policy: The page's settings blocked the loading of a resource at [https://logws1363.ati-host.net/?s=598342&idclient=86d1b7ac-b…5.page&r=1440x900x24x24&re=1432x393&hl=8x34x13&lng=en-GB&x1=[urn:bbc:cps:d1665420-b348-d54b-9e55-3766667eae59]](https://logws1363.ati-host.net/?s=598342&idclient=86d1b7ac-b%E2%80%A65.page&r=1440x900x24x24&re=1432x393&hl=8x34x13&lng=en-GB&x1=%5Burn:bbc:cps:d1665420-b348-d54b-9e55-3766667eae59%5D)
> > * Content Security Policy: The page's settings blocked the loading of a resource at [https://toggles.test.api.bbci.co.uk/toggles?application=simo…_amp_source_origin=https://www.test.bbc.com&geoiplookup=true](https://toggles.test.api.bbci.co.uk/toggles?application=simo%E2%80%A6_amp_source_origin=https://www.test.bbc.com&geoiplookup=true) ("connect-src").
> > 

> 
> Are those errors present on branch `latest` or only this branch?

This issue is now fixed after upgrading to latest version of node..

**Pull request merged**